FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/Dicom/DicomApiDataClient.cs

```diff
@@ -0,0 +1,190 @@
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _dicomApiDataSource = dataSource;
+            _httpClient = httpClient;
+            _httpClient.DefaultRequestHeaders.Add("Accept", "application/dicom+json");
+            _accessTokenProvider = new AzureAccessTokenProvider(
+                tokenCredentialProvider.GetCredential(TokenCredentialTypes.External),
+                diagnosticLogger,
+                new Logger<AzureAccessTokenProvider>(new LoggerFactory()));
+            _diagnosticLogger = diagnosticLogger;
+            _logger = logger;
+        }
+
+        public async Task<string> SearchAsync(
+            BaseApiOptions dicomApiOptions,
+            CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException();
+            }
+
+            Uri searchUri;
+            try
+            {
+                searchUri = CreateSearchUri(dicomApiOptions);
+            }
+            catch (Exception ex)
+            {
+                _diagnosticLogger.LogError(string.Format("Create search Uri failed, Reason: '{0}'", ex.Message));
+                _logger.LogInformation(ex, "Create search Uri failed, Reason: '{reason}'", ex.Message);
+                throw new ApiSearchException("Create search Uri failed", ex);
+            }
+
+            string accessToken = null;
+            if (dicomApiOptions.IsAccessTokenRequired)
+            {
+                try
+                {
+                    if (_dicomApiDataSource.Authentication == AuthenticationType.ManagedIdentity)
+                    {
+                        // Currently we support accessing service endpoints with Managed Identity.
+
+                        // The thread-safe AzureServiceTokenProvider class caches the token in memory and retrieves it from Azure AD just before expiration.
+                        // https://docs.microsoft.com/en-us/dotnet/api/overview/azure/service-to-service-authentication#using-the-library
+                        accessToken = await _accessTokenProvider.GetAccessTokenAsync(DicomApiConstants.DicomResourceUrl, cancellationToken);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _diagnosticLogger.LogError(string.Format("Get server access token failed, Reason: '{0}'", ex.Message));
+                    _logger.LogInformation(ex, "Get server access token failed, Reason: '{reason}'", ex.Message);
+                    throw new ApiSearchException("Get server access token failed", ex);
+                }
+            }
+
+            return await GetResponseFromHttpRequestAsync(searchUri, accessToken, cancellationToken);
+        }
+
+        public string Search(BaseApiOptions serverApiOptions)
+        {
+            return SearchAsync(serverApiOptions).Result;
```

Comment:
If dicom service only supports search async, how about throwing noimplemented exception here?

Reply:
Good point. Quan and I are just discussing about it today. Currently `ApiSearchException` is thrown for the synchronous search in DICOM.